### DISCUSSION TITLE
How come static Constructor is called before onRuntimeInitialized

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#18854] [Q&A]
I have a static class, its constructor is called before  `onRuntimeInitialized`. is it right ?
what should I do to make it call after `onRuntimeInitialized` ?

emcc version 3.1.29
OS: Windows 11

**Comment by sbc100** [comment]
I think the preRun callbacks are run before static constructors, so you could try using `addOnPreRun`.
